sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/JacksonAdapter.java

```diff
@@ -70,7 +81,9 @@ public JacksonAdapter() {
                 .registerModule(AdditionalPropertiesSerializer.getModule(flatteningMapper))
                 .registerModule(AdditionalPropertiesDeserializer.getModule(flatteningMapper))
                 .registerModule(FlatteningSerializer.getModule(simpleMapper()))
-                .registerModule(FlatteningDeserializer.getModule(simpleMapper()));    }
+                .registerModule(FlatteningDeserializer.getModule(simpleMapper()));
+        headerMapper = initializeHeaderMapper(new ObjectMapper());
```

Comment:
jfyi we could also do: 

```
headerMapper  = simpleMapper
    .copy()
    .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```
https://static.javadoc.io/com.fasterxml.jackson.core/jackson-databind/2.9.8/com/fasterxml/jackson/databind/ObjectMapper.html#copy--